sdk/proto/provider.proto

```diff
@@ -146,3 +146,11 @@ message DeleteRequest {
     string urn = 2;                        // the Pulumi URN for this resource.
     google.protobuf.Struct properties = 3; // the current properties on the resource.
 }
+
+// ErrorResourceInitFailed is sent as a Detail `ResourceProvider.{Create, Update}` fail because a
+// resource was created successfully, but failed to initialize.
+message ErrorResourceInitFailed {
+    string id = 1;                         // the ID of the created resource.
```

Comment:
It makes me a tiny bit nervous that we're inlining fields from `{Create,Update}Response` here. Could we just add fields of those types (or a single field with an appropriate union type)?

Reply:
We agreed to add comments to `CreateResponse` and `UpdateResponse`.